src/Perception/src/ArucoDetector.cpp

```diff
@@ -0,0 +1,299 @@
+                                                                                         {"5X5_50",  cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_5X5_50},
+                                                                                         {"5X5_100", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_5X5_100},
+                                                                                         {"5X5_250", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_5X5_250},
+                                                                                         {"5X5_1000",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_5X5_1000},
+                                                                                         {"6X6_50",  cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_6X6_50},
+                                                                                         {"6X6_100", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_6X6_100},
+                                                                                         {"6X6_250", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_6X6_250},
+                                                                                         {"6X6_1000",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_6X6_1000},
+                                                                                         {"7X7_50",  cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_50},
+                                                                                         {"7X7_100", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_100},
+                                                                                         {"7X7_250", cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_250},
+                                                                                         {"7X7_1000",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_7X7_1000},
+                                                                                         {"ARUCO_ORIGINAL",cv::aruco::PREDEFINED_DICTIONARY_NAME::DICT_ARUCO_ORIGINAL} };
+};
+
+ArucoDetector::ArucoDetector() :
+               m_pimpl(std::make_unique<Impl>())
+{
+}
+
+ArucoDetector::~ArucoDetector() = default;
+
+bool ArucoDetector::initialize(std::weak_ptr<IParametersHandler> handler)
+{
+    std::string_view printPrefix = "[ArucoDetector::initialize] ";
+    auto handle = handler.lock();
+    if (handle == nullptr)
+    {
+        log()->error("{} "
+        "The parameter handler has expired. Please check its scope.", printPrefix);
+        return false;
+    }
+
+    std::string dictName;
+    if (!handle->getParameter("marker_dictionary", dictName))
+    {
+        log()->error("{} "
+        "The parameter handler could not find \" marker_dictionary \" in the configuration file.",
+        printPrefix);
+        return false;
+    }
+
+    if (m_pimpl->availableDict.find(dictName) == m_pimpl->availableDict.end())
+    {
+        std::cerr << printPrefix
```

Comment:
Could you please use `log()->error` also here?

Reply:
Thanks, I had missed it. Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/159/commits/81b92b54e3bfd045dc63b7679120089be2f59b7c